### ISSUE TITLE
Adjust Github Actions for Crain's Sanity

### ISSUE THREAD

**Issue by Crain-32** [SuperDude88/TWWHD-Randomizer#39]
I blame Github for their own stupidity on not really having any good examples for Actions. But these are some things I've found that have helped me/my Team when it came to using Actions. And as I was reviewing our Actions I decided sharing is caring.

Think of an overall Action YML in two ways. You have actions that do things or actions that configure things.

I'd recommend if you have a "doing" Action, that it has one Job. For example right now `build_latest.yml` has two Jobs, and they're linked in terms of what you want, but completely removed in terms of actual workflow. <https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow>by using this pattern, we can split the Workflows into two sub folders
```
/.github
   /workflows
      /console
        build_latest.yml
      /desktop
        build_latest.yml
      build_latest.yml
```
And the `/workflow/build_latest.yml` would look like the following.
```yml
name: Build Latest - Joint # Name this whatever

on: 
   workflow_dispatch:
   pull_request:
     types: [merged] # Double check this isn't the wrong one
     
 jobs:
   build-console:
     uses: ./.github/workflows/console/build_latest.yml
     # Can pass in Environment Stuff/Secrets
   build_desktop:
     uses: ./.github/workflows/desktop/build_latest.yml
```

Regarding the usage of `on: pull-request:`, I'd recommend that over a concept of a "nightly" build. Most large projects have Nightly Builds due to the sheer amount of PRs that are merged in each day. Where TWWHDR *might* merge in a PR every day. In that case it is much easier to just have Merged-PR artifacts and download them from the Actions page when needed. Beyond that Nightly Builds are usually on a package manager system, where we're distributing from the Github Releases page. Which if you never remove the nightly builds.... gets hard to navigate fast.

The other nice thing about removing the Nightly approach is now you can use the `new Prerelease` Actions Trigger for the release Action. No need to get a changelist since if you're making a release you can use Github's built-in changelog generator. Then you can append the Artifacts to the Release with an action I can't remember the name of (I know it exists, we use it at work). After all that finishes, it just sets the Pre-Release to Release.

TLDR:
 Modularize the Actions to allow for greater reuse and localized context
 Remove "Nightly" builds in favor of PR-Merge Artifacts.
